### PR TITLE
Return full objects instead of summaries for nested topics in GET topics/broadcasts responses [pr]

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?

Modifies the contentfulEntry helper `getTopicTemplatesFromContentfulEntry` and `getMessageTemplateFromContentfulEntryAndTemplateName` to set the message or template topic by calling `helpers.topic.getById` instead of the `helpers.contentfulEntry.getSummaryFromContentfulId` -- which doesn't include campaign information (which ideally we'd return to avoid making multiple `GET /topics/:id` requests when processing a user's reply to an outbound `askYesNo`-- when looking up the `askYesNo` topic to find replies, if a user says yes, it's convenient to have the campaign id available for posting campaign activity)

#### How should this be reviewed?

* Verify a full topic object is returned in the `message.topic` properties of a `GET /broadcasts` request for a non askYesNo (e.g. `5t2fmKd2iQgaCCy8Kgg0CI`)

```
// 20180815190247
// http://localhost:5000/v1/broadcasts/5t2fmKd2iQgaCCy8Kgg0CI?apiKey=totallysecret&test=123

{
  "data": {
      "id": "5t2fmKd2iQgaCCy8Kgg0CI",
      "name": "Test photoPostBroadcast - Pump it up",
      "type": "photoPostBroadcast",
      "createdAt": "2018-08-09T21:14:49.304Z",
      "updatedAt": "2018-08-13T18:19:04.214Z",
      "message": {
        "text": "Have you posted a flyer yet? Text START to share your photo.",
        "attachments": [
          
        ],
        "template": "photoPostBroadcast",
        "topic": {
          "id": "4xXe9sQqmIeiWauSUu6kAY",
          "name": "Pump It Up - Submit Flyer",
          "type": "photoPostConfig",
          "createdAt": "2018-08-01T14:41:30.242Z",
          "updatedAt": "2018-08-13T13:31:32.583Z",
          "postType": "photo",
          "campaign": {
            "id": 8142,
            "title": "Pump It Up",
            "tagline": "Print & post our tire flyers to keep friends safe on the road.",
            "status": "active",
            "currentCampaignRun": {
              "id": 8161
            },
            "endDate": {
              "date": "2018-09-30 23:59:00.000000",
              "timezone_type": 1,
              "timezone": "-04:00"
            }
          },
          ...
```

* Verify a full topic is returned in the `templates.saidYes.topic` property of a `GET /broadcasts` request for an askYesNo (e.g. `2X4r3fZrTGA2mGemowgiEI`)

```
// http://localhost:5000/v1/broadcasts/2X4r3fZrTGA2mGemowgiEI?apiKey=totallysecret&test=123

{
  "data": {
    "id": "2X4r3fZrTGA2mGemowgiEI",
    "name": "Test askYesNo - Pump it up",
    "type": "askYesNo",
    "createdAt": "2018-08-06T23:34:56.395Z",
    "updatedAt": "2018-08-13T17:45:47.751Z",
    "message": {
      "text": "Join Pump it Up? \n\nYes No",
      "attachments": [
        
      ],
      "template": "askYesNo",
      "topic": {
        
      }
    },
    "templates": {
      "saidYes": {
        "text": "Great! Text START to submit a photo.",
        "topic": {
          "id": "4xXe9sQqmIeiWauSUu6kAY",
          "name": "Pump It Up - Submit Flyer",
          "type": "photoPostConfig",
          "createdAt": "2018-08-01T14:41:30.242Z",
          "updatedAt": "2018-08-13T13:31:32.583Z",
          "postType": "photo",
          "campaign": {
            "id": 8142,
            "title": "Pump It Up",
            "tagline": "Print & post our tire flyers to keep friends safe on the road.",
            "status": "active",
            "currentCampaignRun": {
              "id": 8161
            },
            "endDate": {
              "date": "2018-09-30 23:59:00.000000",
              "timezone_type": 1,
              "timezone": "-04:00"
            }
          },
          "templates": {
          ....
```

#### Relevant tickets
https://www.pivotaltracker.com/story/show/157369418

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [x] Tested on staging.

### CODE REVIEW

File: lib/helpers/contentfulEntry.js [hunk 1]
```diff
@@ -64,11 +64,12 @@ function getSummaryFromContentfulEntry(contentfulEntry) {
 /**
  * @param {Object} contentfulEntry
  * @param {String} templateName
- * @return {Object}
+ * @return {Promise}
```
[the remark this answers is not in the document]
Comment (rapala61) [reply lib/helpers/contentfulEntry.js]: I was under the impression that by using `async` functions, the return was always a `Promise`? In this case the return is an object wrapped in a resolved Promise. What do you suggest the comment should change to?

File: lib/helpers/topic.js [hunk 2]
```diff
@@ -203,48 +211,53 @@ function parseTopicFromContentfulEntry(contentfulEntry) {
...
-        data.campaign = { id: campaignId, status: 'closed' };
-        data.templates = module.exports
-          .parseTopicTemplatesFromContentfulEntryAndCampaign(contentfulEntry, null);
-        return data;
-      }
+  try {
+    data.campaign = campaignConfigEntry ? await helpers.campaign
+      .getById(contentful.getCampaignIdFromContentfulEntry(campaignConfigEntry)) : {};
+  } catch (error) {
+    // The campaignId is set via freeform text field on the campaign content type.
```
Comment (rapala61) [inline lib/helpers/topic.js]: This line can be improved by being more explicit about where this things are happening, example: `The campaignId value is set via manual input on the campaign content type entry using the Contentful UI`

File: lib/helpers/contentfulEntry.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaronschachter) [reply lib/helpers/contentfulEntry.js]: I was under the same impression as @rapala61. Looking at http://usejsdoc.org/tags-async.html, and this looks like a clean way to define the return... in our case here it'd be a `@return {Promise<Object>}` (to keep consistent with using uppercase names for type returned)

```
/**
 * Download data from the specified URL.
 *
 * @async
 * @function downloadData
 * @param {string} url - The URL to download from.
 * @return {Promise<string>} The data from the URL.
 */ 
```

File: lib/helpers/topic.js [hunk 3]
```diff
@@ -259,9 +272,9 @@ module.exports = {
   getFieldValueFromContentfulEntryAndTemplateName,
   getPostTypeFromContentType,
   getTemplateInfoFromContentfulEntryAndTemplateName,
+  getTemplatesWithDefaultsFromContentfulEntryAndCampaign,
```
[the remark this answers is not in the document]
Comment (aaronschachter) [reply lib/helpers/topic.js]: I'm liking these suggestions of shorter named functions... @rapala61 sounds good on your end?

**Pull request merged**